GVFS/GVFS.Virtualization/Projection/GitIndexProjection.GitIndexParser.cs

```diff
@@ -105,18 +113,30 @@ private FileSystemTaskResult AddToProjection(GitIndexEntry data)
                 return FileSystemTaskResult.Success;
             }
 
-            private FileSystemTaskResult AddToModifiedFiles(GitIndexEntry data)
+            private FileSystemTaskResult AddToModifiedFiles(
+                GitIndexEntry data,
+                Dictionary<string, PlaceholderListDatabase.PlaceholderData> placeholderDataByPath,
+                List<PlaceholderListDatabase.PlaceholderData> placeholderDataToKeep)
             {
+                data.ParsePath();
+
                 if (!data.SkipWorktree)
                 {
                     // A git command (e.g. 'git reset --mixed') may have cleared a file's skip worktree bit without
                     // triggering an update to the projection.  Ensure this file is in GVFS's modified files database
-                    data.ParsePath();
                     return this.projection.AddModifiedPath(data.GetFullPath());
                 }
                 else
                 {
                     data.ClearLastParent();
+
+                    string placeholderPath = data.GetFullPath();

```

Comment:
Make sure that on both platforms we end up with the right path separators (and add a functional test that will fail on Windows without this fix).